internal/driver/registry_factory.go

```diff
@@ -62,23 +71,63 @@ func NewDefaultRegistry(ctx context.Context, flags *pflag.FlagSet, withoutNetwor
 	return r, nil
 }
 
-func NewSqliteTestRegistry(t testing.TB, debugOnDisk bool) *RegistryDefault {
+func NewSqliteTestRegistry(t testing.TB, debugOnDisk bool, opts ...TestRegistryOption) *RegistryDefault {
 	mode := dbx.SQLiteMemory
 	if debugOnDisk {
 		mode = dbx.SQLiteDebug
 	}
-	return NewTestRegistry(t, dbx.GetSqlite(t, mode))
+	return NewTestRegistry(t, dbx.GetSqlite(t, mode), opts...)
 }
 
-type newRegistryOption func(t testing.TB, r *RegistryDefault)
+type TestRegistryOption func(t testing.TB, r *RegistryDefault)
 
-func WithNamespaces(namespaces []*namespace.Namespace) newRegistryOption {
+func WithNamespaces(namespaces []*namespace.Namespace) TestRegistryOption {
 	return func(t testing.TB, r *RegistryDefault) {
 		require.NoError(t, r.c.Set(config.KeyNamespaces, namespaces))
 	}
 }
+func WithGRPCUnaryInterceptors(i ...grpc.UnaryServerInterceptor) TestRegistryOption {
+	return func(_ testing.TB, r *RegistryDefault) {
+		r.defaultUnaryInterceptors = i
+	}
+}
+
+type selfSignedCert struct {
+	once sync.Once
+	cert *tls.Certificate
+	err  error
+}
+
+var sharedTestCert selfSignedCert
+
+func (s *selfSignedCert) generate() {
+	s.once.Do(func() {
+		key, err := ecdsa.GenerateKey(elliptic.P521(), rand.Reader)
```

Comment:
Would it be faster to use instead?
```suggestion
		key, err := ecdsa.GenerateKey(elliptic.P224(), rand.Reader)
```

Reply:
Some quick benchmarking suggests we should go with P256.

```
goos: linux
goarch: amd64
pkg: github.com/ory/keto/internal/driver
cpu: 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
BenchmarkCertificateGeneration/P256-8         	    6140	    188965 ns/op	   37140 B/op	     866 allocs/op
BenchmarkCertificateGeneration/P224-8         	    2575	    457371 ns/op	   39339 B/op	     927 allocs/op
BenchmarkCertificateGeneration/P384-8         	     705	   1803005 ns/op	   43231 B/op	     931 allocs/op
BenchmarkCertificateGeneration/P521-8         	     212	   5367458 ns/op	   47601 B/op	     936 allocs/op
```